adapter/test/stdio/stdio_adapter_basic_test.cpp

```diff
@@ -0,0 +1,1293 @@
+      REQUIRE(size_written == request_size);
+      if (biggest_size_written < request_size)
+        biggest_size_written = request_size;
+    }
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+    REQUIRE(fs::file_size(info.new_file) == biggest_size_written);
+  }
+
+  SECTION("write to new file") {
+    FILE* fd = fopen(info.new_file.c_str(), "w+");
+    REQUIRE(fd != nullptr);
+    size_t total_size_written = 0;
+    for (size_t i = 0; i < info.num_iterations; ++i) {
+      size_t request_size =
+          args.request_size + (rand_r(&info.offset_seed) % args.request_size);
+      std::string data(request_size, '1');
+      size_t size_written =
+          fwrite(data.c_str(), sizeof(char), request_size, fd);
+      REQUIRE(size_written == request_size);
+      total_size_written += size_written;
+    }
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+    REQUIRE(fs::file_size(info.new_file) == total_size_written);
+  }
+  posttest();
+}
+
+TEST_CASE("BatchedReadSequentialRSVariable",
+          "[process=" + std::to_string(info.comm_size) +
+              "]"
+              "[operation=batched_read]"
+              "[request_size=type-variable][repetition=1024]"
+              "[pattern=sequential][file=1]") {
+  pretest();
+  SECTION("read from existing file") {
+    FILE* fd = fopen(info.existing_file.c_str(), "r");
+    REQUIRE(fd != nullptr);
+    std::string data(args.request_size, '1');
+    size_t current_offset = 0;
+    for (size_t i = 0; i < info.num_iterations; ++i) {
+      size_t request_size = (args.request_size +
+                             (rand_r(&info.offset_seed) % args.request_size)) %
+                            (info.total_size - current_offset);
+      std::string data(request_size, '1');
+      size_t size_read = fread(data.data(), sizeof(char), request_size, fd);
+      REQUIRE(size_read == request_size);
+      current_offset += size_read;
+    }
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+  }
+
+  SECTION("read from existing file always at start") {
+    FILE* fd = fopen(info.existing_file.c_str(), "r");
+    REQUIRE(fd != nullptr);
+
+    for (size_t i = 0; i < info.num_iterations; ++i) {
+      int status = fseek(fd, 0, SEEK_SET);
+      REQUIRE(status == 0);
+      size_t offset = ftell(fd);
+      REQUIRE(offset == 0);
+      size_t request_size =
+          args.request_size + (rand_r(&info.offset_seed) % args.request_size);
+      std::string data(request_size, '1');
+      size_t size_read = fread(data.data(), sizeof(char), request_size, fd);
+      REQUIRE(size_read == request_size);
+    }
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+  }
+  posttest();
+}
+
+TEST_CASE("BatchedReadRandomRSVariable",
+          "[process=" + std::to_string(info.comm_size) +
+              "]"
+              "[operation=batched_read]"
+              "[request_size=type-variable]"
+              "[repetition=1024][pattern=random][file=1]") {
```

Comment:
Should `repetition` be bound to `info.num_iterations`, or does the 1024 come from somewhere else?

Reply:
added it as a variable to tags.